src/en/wip-storage.md

```diff
@@ -0,0 +1,212 @@
+
+## Deploying a charm with storage requirements
+
+For the purposes of demonstration, we will use a charm which has been modified to support storage:
+https://code.launchpad.net/~axwalk/charms/trusty/postgresql/trunk
+
+### Preparing storage
+By default, charms with storage requirements will allocate those resources on the root filesystem of the unit where they are deployed.
+To make use of additional storage resources, Juju needs to know what they are. Some providers (eg EC2) support generic default storage pools (see #REF  the documentation on provider support), but in the case of no default support or a desire to be more specific, Juju has the `juju storage` command and subcommands to create and manage storage resources
+
+juju storage 
+juju storage list
+juju storage pool list
+juju storage pool create
+juju storage volume list
+
+
+
+### deploying with storage constraints
+
+As previously mentioned, a charm which requires storage will automatically allocate the default storage (unit filesystem) by default. It is possible to instead specify the storage to be used when the service is deployed, using constraints.
+
+The constraints can specify the type/pool and size of the storage required.
+
+```
+juju deploy <charm> --storage <label>=<pool>,<size>
+```
+
+For example, to deploy the postgresql service and have it use the unit’s local filesystem for 10 Gigabytes of storage for its ‘data’ storage requirement:
+
+```
+juju deploy cs:~axwalk/postgresql --storage data=rootfs,10G
+```
+
+We can also deploy using a local loopback mount
+
+```
+juju deploy cs:~axwalk/postgresql --storage data=loop,5G
+```
+
+If the size is omitted...
+
+```
+juju deploy cs:~axwalk/postgresql --storage data=rootfs
+```
+
+Juju will use a default size of 1G, unless the charm itself has specified a minimum value, in which case that will be used.
+
+When deploying on a provider which supplies storage, the supported storage pool types may be used in addition to ‘loop’ and ‘rootfs’. For example, on using Amazon’s EC2 provider, we can make use of the default ‘ebs’ storage pool
+
+```
+juju deploy cs:~axwalk/postgresql --storage data=ebs,10G
+```
+
+Cloud providers may support more than one type of storage. For example, in the case of EC2, we can also make use of the ebd-ssd pool, which is SSD-based storage, and hence faster and better for some storage requirements.
+
+```
+juju deploy cs:~axwalk/postgresql --storage data=ebs-ssd
+```
+
+We can also merely specify the size, in which case Juju will use the default pool for the selected environment.  E.g. 
+
+```
+juju deploy cs:~axwalk/postgresql --storage data=10G
+```
+
+Which, on the EC2 provider, will create a 10 gigabyte volume in the ‘ebs’ pool.
+
+Charms may specify as many different types of storage as they require, in which case they may all be specified using the constraint, or some or all can be omitted to accept the default values:
```

Comment:
Instead "Charms may declare multiple types* of storage..."

Don't really like types, but okay for now.